src/multiplayer_server_scanner.cpp

```diff
@@ -15,6 +16,7 @@ ServerScanner::~ServerScanner()
 
 void ServerScanner::scanMasterServer(string url)
 {
+    abort_wait.notify_all();
```

Comment:
This isn't the right spot. While it is harmless in how it is used, the line below is to protect `scanMasterServer` from being multiple times, not to actually abort it. (`joinable` means "thread is running", c++ can be wierd like that)

Reply:
True, but if I'm not mistaken, it causes the serverlist to immediately try to refresh when scanMasterServer is requested. It's not so much about telling the thread to stop and more to refresh faster when the UI wants to show it.

I can still remove it if you like

Reply:
Speaking of which, shouldn't master_server_url still be set if the thread is running? EE only has one server list URL, so it doesn't really matter. Just something I just noticed if there ever were to be more than one URL

Reply:
Well, it's currently also doing something still slightly "bad", it's accessing `master_server_url` from two threads without a lock, which could cause crashes. Chances of it happening are really slim however, due to the relatively low poll rate.